src/modules/recurringMessage/recurringMessage.helpers.ts

```diff
@@ -134,9 +139,37 @@ export const listRecurringMessages = async (interaction: ChatInputCommandInterac
 export const relaunchRecurringMessages = async (client: Client<true>) => {
   const recurringMessages = await cache.get('recurringMessages', []);
 
-  recurringMessages.forEach(({ id, channelId, frequency, message }) => {
+  const channelsToClear = new Set<string>();
+  for (const { id, channelId, frequency, message } of recurringMessages) {
+    if (!client.channels.cache.get(channelId)) {
+      channelsToClear.add(channelId);
+      continue;
+    }
     const job = createRecurringMessage(client, channelId, frequency, message);
     job.start();
     inMemoryJobList.push({ id, job });
+  }
+
+  await cache.set(
+    'recurringMessages',
+    recurringMessages.filter(({ channelId }) => !channelsToClear.has(channelId)),
+  );
+};
+
+export const removeAllFromChannel = async (channel: DMChannel | NonThreadGuildBasedChannel) => {
+  const { id } = channel;
+
+  const recurringMessages = await cache.get('recurringMessages', []);
+  const jobsToRemove = recurringMessages.filter(({ channelId }) => id === channelId);
+
+  await cache.set(
+    'recurringMessages',
+    recurringMessages.filter(({ channelId }) => id !== channelId),
+  );
+
+  jobsToRemove.forEach(({ id }) => {
+    const job = inMemoryJobList.find(({ id: jobId }) => id === jobId)?.job;
+    if (!job) return;
+    job.stop();
```

Comment:
why did you choose to go with a `forEach` vs what you did previously with a `for of` ?

Reply:
no async and no break or continue here

Reply:
this is a `break`

https://github.com/codinglab-io/discord-bot/blob/c9bbfddbe665eee1edcd16a31fd58d304e2534b8/src/modules/recurringMessage/recurringMessage.helpers.ts#L172